### PR TITLE
Validate id equals name

### PR DESCRIPTION
**Pull request by reutshal**
Added this validation to: integrations, scripts, **new** playbooks
fixes : https://github.com/demisto/etc/issues/19275

### CODE REVIEW

File: demisto_sdk/validation/file_validator.py [hunk 1]
```diff
@@ -310,9 +311,15 @@ def validate_added_files(self, added_files):
                 if self.id_set_validator.is_file_has_used_id(file_path):
                     self._is_valid = False
 
-            if re.match(TEST_PLAYBOOK_REGEX, file_path, re.IGNORECASE):
-                if not self.conf_json_validator.is_test_in_conf_json(collect_ids(file_path)):
+            returned_playbook_regex = checked_type(file_path, PLAYBOOKS_REGEXES_LIST, re.IGNORECASE)
```
Comment (ronykoz) [inline demisto_sdk/validation/file_validator.py]: don't think we need the re.IGNORECASE here, should be True

File: demisto_sdk/common/hook_validations/playbook.py [hunk 2]
```diff
@@ -8,9 +8,22 @@ def is_valid_version(self):
         # type: () -> bool
         return self._is_valid_version()
 
-    def is_valid_playbook(self):  # type: () -> bool
+    def is_valid_playbook(self, is_new_playbook):  # type: (bool) -> bool
```
Comment (ronykoz) [inline demisto_sdk/common/hook_validations/playbook.py]: We should have a default value for the arg

File: demisto_sdk/common/hook_validations/playbook.py [hunk 3]
```diff
@@ -8,9 +8,22 @@ def is_valid_version(self):
...
-            self.is_valid_version()
-        ]
-        return all(answers)
+        if is_new_playbook:
+            new_playbook_checks = [
+                self.is_valid_version(),
+                self.is_id_equals_name()
+            ]
+            answers = all(new_playbook_checks)
+        else:
```
Comment (ronykoz) [inline demisto_sdk/common/hook_validations/playbook.py]: add a comment explaining this

File: demisto_sdk/common/hook_validations/base_validator.py [hunk 4]
```diff
@@ -100,3 +101,18 @@ def is_subset_dictionary(new_dict, old_dict):
             if arg not in old_dict.keys() and required:
                 return False
         return True
+
+    def _is_id_equals_name(self, file_type):
```
Comment (ronykoz) [inline demisto_sdk/common/hook_validations/base_validator.py]: are we using file_type elsewhere?

File: demisto_sdk/common/hook_validations/base_validator.py [hunk 5]
```diff
@@ -100,3 +101,18 @@ def is_subset_dictionary(new_dict, old_dict):
...
+        """Validate that the id of the file equals to the name."""
```
Comment (ronykoz) [inline demisto_sdk/common/hook_validations/base_validator.py]: add args and returns sections

File: demisto_sdk/common/hook_validations/base_validator.py (hunk 4 above)
Comment (ronykoz) [inline demisto_sdk/common/hook_validations/base_validator.py]: how are we validating json files?

Comment (ronykoz) [inline demisto_sdk/common/hook_validations/base_validator.py]: jsons don't get here, right?

**Comment by ronykoz** [comment]
update from master and add changelog too please

File: demisto_sdk/common/hook_validations/base_validator.py (hunk 4 above)
Comment (reutshal) [reply demisto_sdk/common/hook_validations/base_validator.py]: right. this function is only called from script, integration or playbook validator

File: demisto_sdk/common/hook_validations/base_validator.py (hunk 5 above)
Comment (reutshal) [reply demisto_sdk/common/hook_validations/base_validator.py]: Done

File: demisto_sdk/common/hook_validations/playbook.py (hunk 2 above)
Comment (reutshal) [reply demisto_sdk/common/hook_validations/playbook.py]: Set to True

File: demisto_sdk/common/hook_validations/playbook.py (hunk 3 above)
Comment (reutshal) [reply demisto_sdk/common/hook_validations/playbook.py]: Done

File: demisto_sdk/validation/file_validator.py (hunk 1 above)
Comment (reutshal) [reply demisto_sdk/validation/file_validator.py]: removed this line, see the updated file

File: demisto_sdk/common/hook_validations/playbook.py [hunk 6]
```diff
@@ -8,9 +8,26 @@ def is_valid_version(self):
         # type: () -> bool
         return self._is_valid_version()
 
-    def is_valid_playbook(self):  # type: () -> bool
+    def is_valid_playbook(self, is_new_playbook=True):  # type: (bool) -> bool
         """Check whether the playbook is valid or not"""
```
Comment (ronykoz) [inline demisto_sdk/common/hook_validations/playbook.py]: add args + returns to the docstring

File: demisto_sdk/common/hook_validations/playbook.py [hunk 7]
```diff
@@ -8,9 +8,26 @@ def is_valid_version(self):
...
+        # for new playbooks - run all playbook checks.
+        # for modified playbooks - only validate the version.
+
+        if is_new_playbook:
+            new_playbook_checks = [
+                self.is_valid_version(),
+                self.is_id_equals_name()
+            ]
+            answers = all(new_playbook_checks)
+        else:
```
Comment (ronykoz) [inline demisto_sdk/common/hook_validations/playbook.py]: move the comment to here

File: demisto_sdk/common/hook_validations/playbook.py (hunk 6 above)
Comment (reutshal) [reply demisto_sdk/common/hook_validations/playbook.py]: Done

File: demisto_sdk/common/hook_validations/playbook.py (hunk 7 above)
Comment (reutshal) [reply demisto_sdk/common/hook_validations/playbook.py]: Done

File: demisto_sdk/common/hook_validations/base_validator.py [hunk 8]
```diff
@@ -99,3 +100,24 @@ def is_subset_dictionary(new_dict, old_dict):
...
                 return False
         return True
+
+    def _is_id_equals_name(self, file_type):
+        """Validate that the id of the file equals to the name.
+         Args:
+            file_type (str): the file type. can be 'integration', 'script' or 'playbook'
+
+        Returns:
+            bool. Whether the file id equals to its name
```
[the remark this answers is not in the document]
Comment (reutshal) [reply demisto_sdk/common/hook_validations/base_validator.py]: Done

File: demisto_sdk/common/hook_validations/base_validator.py [hunk 9]
```diff
@@ -99,3 +100,24 @@ def is_subset_dictionary(new_dict, old_dict):
...
+        """
+        file_id = ''
+        if file_type in ID_IN_ROOT:
+            file_id = self.current_file.get('id')
+        elif file_type in ID_IN_COMMONFIELDS:
+            file_id = self.current_file.get('commonfields', {}).get('id')
```
[the remark this answers is not in the document]
Comment (reutshal) [reply demisto_sdk/common/hook_validations/base_validator.py]: Done

File: demisto_sdk/common/hook_validations/integration.py [hunk 10]
```diff
@@ -37,14 +38,14 @@ def is_backward_compatible(self):
     def is_valid_file(self, validate_rn=True):
         # type: (bool) -> bool
         """Check whether the Integration is valid or not"""
-        super(IntegrationValidator, self).is_valid_file(validate_rn)
         answers = [
```
[the remark this answers is not in the document]
Comment (guykeller) [reply demisto_sdk/common/hook_validations/integration.py]: Not sure I agree with that one, I think we should eventually change it everywhere, starting with this file.
What do you think? @ronykoz

File: demisto_sdk/common/constants.py [hunk 11]
```diff
@@ -330,6 +330,14 @@ def no_common_server_python(path):
 REPORT_REGEX = r'{}{}.*report-.*\.json$'.format(CAN_START_WITH_DOT_SLASH, REPORTS_DIR)
 MISC_REPUTATIONS_REGEX = r'{}{}.reputations.json$'.format(CAN_START_WITH_DOT_SLASH, MISC_DIR)
 
+ID_IN_COMMONFIELDS = [  # entities in which 'id' key is under 'commonfields'
+    'integration',
+    'script'
+]
+ID_IN_ROOT = [  # entities in which 'id' key is in the root
+    'playbook'
+]
```
[the remark this answers is not in the document]
Comment (guykeller) [reply demisto_sdk/common/constants.py]: They don't have to be constants of the class.
They're only used once.
Use a function that checks directly if if file type is in `['script', 'integration']` or is "playbook"

File: demisto_sdk/common/hook_validations/base_validator.py [hunk 12]
```diff
@@ -99,3 +100,24 @@ def is_subset_dictionary(new_dict, old_dict):
...
+
+        name = self.current_file.get('name', '')
+        if file_id != name:
+            print_error("The ID of the {} '{}' should be equal to its name, "
+                        "please update the file.".format(file_type, self.file_path))
```
Comment (guykeller) [inline demisto_sdk/common/hook_validations/base_validator.py]:
```suggestion
                        "please update the file (path to file: {2}).".format(name, file_id, self.file_path))
```

File: demisto_sdk/validation/file_validator.py [hunk 13]
```diff
@@ -232,6 +232,8 @@ def validate_modified_files(self, modified_files):  # noqa: C901
 
             structure_validator = StructureValidator(file_path, old_file_path)
             if not structure_validator.is_valid_file():
+                if re.match(TEST_PLAYBOOK_REGEX, file_path, re.IGNORECASE):
+                    continue
```
Comment (ronykoz) [inline demisto_sdk/validation/file_validator.py]: better to move this if to be before the is_valid_file

File: demisto_sdk/validation/file_validator.py [hunk 14]
```diff
@@ -320,6 +322,8 @@ def validate_added_files(self, added_files):
 
             structure_validator = StructureValidator(file_path)
             if not structure_validator.is_valid_file():
+                if re.match(TEST_PLAYBOOK_REGEX, file_path, re.IGNORECASE):
```
Comment (ronykoz) [inline demisto_sdk/validation/file_validator.py]: same here

File: demisto_sdk/common/constants.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (reutshal) [reply demisto_sdk/common/constants.py]: @ronykoz thinks it's better to leave them in constants file

File: demisto_sdk/common/hook_validations/integration.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (ronykoz) [reply demisto_sdk/common/hook_validations/integration.py]: lets finish this PR and then do the change across the sdk

File: demisto_sdk/common/hook_validations/playbook.py (hunk 6 above)
Comment (reutshal) [reply demisto_sdk/common/hook_validations/playbook.py]: Done

**Pull request merged**